src/test/java/io/vertx/core/loadbalancing/LoadBalancingTest.java

```diff
@@ -35,6 +36,11 @@ public void testRoundRobin() throws Exception {
     assertEquals(0, selector.selectEndpoint(metrics));
     assertEquals(1, selector.selectEndpoint(metrics));
     assertEquals(2, selector.selectEndpoint(metrics));
+
+    metrics = new ArrayList<>();
```

Comment:
I don't know if there is a way to factorize these special cases checks for all policies.

Reply:
Yes, you could use a separate class with parameterized test junit runner

Reply:
I'll give that a try, thanks.